src/services/textChanges.ts

```diff
@@ -1238,9 +1238,10 @@ namespace ts.textChanges {
                 }
 
                 case SyntaxKind.ImportDeclaration:
+                    const isFirstImport = sourceFile.imports.length && node === first(sourceFile.imports).parent || node === find(sourceFile.statements, isImportDeclaration);

```

Comment:
When would

```
sourceFile.imports.length && node === first(sourceFile.imports).parent
```

be true, but not the following?

```ts
node === find(sourceFile.statements, isImportDeclaration)
```

Reply:
@DanielRosenwasser 

> When would sourceFile.imports.length && node === first(sourceFile.imports).parent


```ts
import { a } from "./b";
import { b } from "./c";
```
when each import statemnt has path

> be true, but not the following? node === find(sourceFile.statements, isImportDeclaration)

```ts
import a from "";
import { a } from "./b";
import { b } from "./c";
```

when there is an import statement with the empty path

Reply:
It sounds like it won't work correctly when

```ts
import { a } from "./b";
// try to remove the following import
import a from "";
import { b } from "./c";
```

Does that still exhibit the same error?

Reply:
@DanielRosenwasser Sorry for the delay with the response, sometimes different time zones matter :). There is a test for a similar case only with leading comment 

![1](https://user-images.githubusercontent.com/509265/74406559-aadd8200-4e38-11ea-869c-a26cbb645dce.png)
![2](https://user-images.githubusercontent.com/509265/74406557-aa44eb80-4e38-11ea-90b0-5556ee49761e.png)
![3](https://user-images.githubusercontent.com/509265/74406554-a9ac5500-4e38-11ea-929b-d43a87b46c8f.png)

Is it an answer to your question?